pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -581,3 +599,137 @@ func installNetworkPolicies(instance *servingv1alpha1.KnativeServing) error {
 	}
 	return nil
 }
+
+func exposeMetricToServiceMonitor(instance *servingv1alpha1.KnativeServing) error {
+	// namespace where knative-serving-operator has been installed
+	namespace, err := getNamespace()
+	if err != nil {
+		log.Info(fmt.Sprintf("empty %s because operator dont want to expose metric or running locally", knativeServingInstalledNamespace))
+		return nil
+	}
+
+	log.Info("exposing metric for installed knative version")
+	knativeVersion.WithLabelValues(version.Version, namespace, instance.GetName()).Set(float64(time.Now().Unix()))
+
+	// Get service knative-serving-operator in order to create service monitor
+	svcDetail, err := getOperatorService(namespace)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			log.Info(fmt.Sprintf("service knative-serving-operator does not exist in %s namespace", namespace))
+			return nil
+		}
+		return err
+	}
+
+	// Add label openshift.io/cluster-monitoring to namespace
+	if err = addMonitoringLabelToNamespace(namespace); err != nil {
+		return err
+	}
+
+	// CreateServiceMonitors creates ServiceMonitors objects based on Service objects.
+	if _, err := metric.CreateServiceMonitors(cfg, namespace, []*v1.Service{svcDetail}); err != nil {
+		if strings.Contains(err.Error(), "already exists") {
+			log.Info(fmt.Sprintf("ServiceMonitor already exist for %s service", svcDetail.Name))
+			return nil
+		}
+		log.Error(err, "failed to create ServiceMonitor so removing added label from namespace")
+		if err = removeMonitoringLabelFromNamespace(namespace); err != nil {
```

Comment:
If other ServiceMonitor is deployed and the label had already been existed, this func removes the label? I wonder that we should remove the label, only when `addMonitoringLabelToNamespace` newly created it.

Reply:
`addMonitoringLabelToNamespace` will add label to the namespace where operator is installed so in that case `removeMonitoringLabelFromNamespace` will remove if there is failure in servicemonitor creation 

In this scenario no other service monitor will be created.

Reply:
Ah, no I mean, for example, if you deployed serving-operator on `openshift-monitoring` ns and it failed, the label will be removed. But `openshift-monitoring` ns still needs the label because other ServiceMonitors use it. (please refer to `$ oc get servicemonitor -n openshift-monitoring`)
This scenario does not happen?

Reply:
I'd ditch the cleanup logic. The label doesn't hurt, does it?

Reply:
Correct. The label does not hurt. So I agree that we ditch the clean up logic.